File: Data Structures & Algorithms/climbing-stairs/submission-1.java
class Solution {
    
    
  
    public int climbStairs(int n) {
        int dp[] = new int[31];
        if(dp[n]!=0){
            return dp[n];
        }
        if(n<=2){
            return n;
        }
         return dp[n]= climbStairs(n-1)+climbStairs(n-2);

        
    }
    
}

File: Data Structures & Algorithms/valid-palindrome-ii/submission-1.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        l,r=0,len(s)-1
        while l<r:
            if s[l]!=s[r]:
                return self.checkPallindrom(s,l,r-1) or self.checkPallindrom(s,l+1,r)
            l+=1
            r-=1
        return True
    

    def checkPallindrom(self,s:str,l:int,r:int)->bool:
        while l<r:
            if s[l]!=s[r]:
                return False
            l+=1
            r-=1
        return True
        

        